File: 1.两数之和/1.两数之和.php
<?php
/*
 * @lc app=leetcode.cn id=1 lang=php
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $arrTarget = array();
        for ($i = 0; $i < count($nums); $i++) {
            $subTarget = $target - $nums[$i];
            //询问有没有人约我
            if (isset($arrTarget[$nums[$i]])) {
                return [ $arrTarget[$nums[$i]], $i ];
            } else {
                //没有人约,写下自己的要求
                $arrTarget[$subTarget] = $i;
            }
        }
        return [];
    }
}
// @lc code=end

File: 1.两数之和/1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	numsMap := map[int]int{}
	for key,value := range nums {
		sub := target - value
		if anotherValue, ok := numsMap[sub]; ok {
			return []int{anotherValue, key}
		}
		numsMap[value] = key
	}
	return []int{}
}
// @lc code=end

//func main(){
////	res := twoSum([]int{3, 3}, 6)
////	fmt.Println(res)
////}
